core/config/project_settings.cpp

```diff
@@ -1428,6 +1429,18 @@ void ProjectSettings::get_argument_options(const StringName &p_function, int p_i
 }
 #endif
 
+void ProjectSettings::set_editor_setting_override(const String &p_setting, const Variant &p_value) {
+	set_setting(EDITOR_SETTING_OVERRIDE_PREFIX + p_setting, p_value);
+}
+
+bool ProjectSettings::has_editor_setting_override(const String &p_setting) const {
+	return has_setting(EDITOR_SETTING_OVERRIDE_PREFIX + p_setting);
+}
```

Comment:
Bit concerned about the potential performance drawbacks of this, but most of the code should be able to handle this? It's good practice to cache the project settings value, but it's not done always.

CC @lawnjelly because of https://github.com/godotengine/godot/pull/103763 (although that tackles 3.x)